src/org/loklak/api/search/GenericScraper.java

```diff
@@ -97,22 +99,47 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
         Query post = RemoteAccess.evaluate(request);
 
         String url = post.get("url", "");
-        String type = post.get("type", "");
+        String type = post.get("type", "default");
 
         URL qurl = new URL(url);
 
         // This can also be done in one line:
         JSONObject genericScraperData = new JSONObject(true);
-        if ("article".equals(type)) {
-            genericScraperData = articleAPI(url, genericScraperData);
-            genericScraperData.put("type", type);
-            printJSON(response, genericScraperData);
-        } else {
-        	genericScraperData.put("error", "Please mentione type of scraper: <article>");
-        	printJSON(response, genericScraperData);
+        switch(type) {
+            // Returns articles text in the webpage
+            case "article":
+                genericScraperData = bpipeExtract(url, genericScraperData, CommonExtractors.ARTICLE_EXTRACTOR);
+                printJSON(response, genericScraperData);
+                break;
+            // Returns default main text in the webpage
+            case "default":
+                genericScraperData = bpipeExtract(url, genericScraperData, CommonExtractors.DEFAULT_EXTRACTOR);
+                printJSON(response, genericScraperData);
+                break;
+            // Returns big articles text in the webpage
+            case "main":
+                genericScraperData = bpipeExtract(url, genericScraperData, CommonExtractors.LARGEST_CONTENT_EXTRACTOR);
+                printJSON(response, genericScraperData);
+                break;            
+            // Returns all text in the webpage
+            case "all":
+                genericScraperData = bpipeExtract(url, genericScraperData, CommonExtractors.KEEP_EVERYTHING_EXTRACTOR);
+                printJSON(response, genericScraperData);
+                break;            
+            // Returns text in the webpage, based on trained data(see boilerpipe docs)
+            case "canola":
+                genericScraperData = bpipeExtract(url, genericScraperData, CommonExtractors.CANOLA_EXTRACTOR);
+                printJSON(response, genericScraperData);
+                break;            
+            // Returns error message for wrong type
+            default:
+                type = "error";
+                genericScraperData.put("error", "Please mention type of scraper: <type> ('article', 'default', 'main', 'all', 'canola')");
+            	printJSON(response, genericScraperData);
+                break;
```

Comment:
It'd be a nice thing to have these extractors within loklak and then use it. I think to perform this patch efficiently we need to either merge the boilerpipe code with our existing structure and then work on it, thereby leaving us the `CommonExtractors` to work with and enhance.